striptease/data_interface.py

```diff
@@ -23,65 +22,65 @@ def load_sci_data(fname_or_data, module, polarimeter, detector, data_type):
     import datetime
     import os.path
 
-    
     # Test inputs
-    
+
     # File exists?
-    if type(fname_or_data) == str:
-        if os.path.exists(fname_or_data) == False:
-            msg = 'ERROR: File %s does not exist' % fname_or_data
-            print(msg)
-            return -1, -1
-        
-    # fname of correct type?
-    if ((type(fname_or_data) != str) and (str(type(fname_or_data)) != "<class 'h5py._hl.dataset.Dataset'>")):
-        msg = 'ERROR: type(fname) %s is wrong' % str(type(fname_or_data))
-        print(msg)
-        return -1, -1
-        
-    
-    # Module exists?
-    modules = ['O', 'Y', 'R', 'G', 'B', 'R', 'W', 'o', 'y', 'r', 'g', 'b', 'r', 'w']
-    if (module in modules) == False:
-        msg = 'ERROR: Module %s does not exist' % module
-        print(msg)
-        return -1, -1
-    module = module.upper()
-    
+    try:
+        f = open(fname_or_data)
+    except IOError as e:
+        if str(type(fname_or_data)) != "<class 'h5py._hl.dataset.Dataset'>":
```

Comment:
Now I realize I don't understand what a `h5py._hl.dataset.Dataset` type is… If I open a HDF5 file with the following line:

```python
import h5py
f = h5py.open("myfile.h5", "r")
isinstance(f,  h5py._hl.files.File)  # Prints True
```

I would have expected this function to accept variable `f` as it first argument, but it's of a `File` type, not a `Dataset` type. How do you create a dataset?

Reply:
@planck2007, once you explain this point and mark the other comments as «resolved», I can merge this PR. Thanks!

Reply:
> I would have expected this function to accept variable f as it first argument, but it's of a File type, not a Dataset type. How do you create a dataset?

I have no idea of how the dataset is created